### ISSUE TITLE
printf does not support wchar_t

### ISSUE THREAD

**Issue by timblechmann** [fmtlib/fmt#721]
`printf` doesn't seem to work with wchar_t, while `sprintf` does:

```
#include <fmt/printf.h>

int main()
{
  fmt::sprintf(L"Elapsed time: %.2f seconds", 1.23); // compiles
  fmt::printf(L"Elapsed time: %.2f seconds", 1.23);  // does not compile
}
```

**Comment by vitaut** [comment]
It should be pretty straightforward to implement `wprintf`, it's just noone has done it yet. Could you by any chance submit a PR?

**Comment by vitaut** [comment]
> my main use-case disappeared

If you don't have a use case for this feature, then I suggest punting on it for now and revisiting once it's actually needed.

**Comment by vitaut** [comment]
Fixed in #724.
